openfe/setup/atom_mapping/rdfmcs_mapper.py

```diff
@@ -0,0 +1,156 @@
+                  element_change=True):
+    """Take a heavy atom mapping and add Hydrogens"""
+    molA = mapping.molA.to_rdkit()
+    molB = mapping.molB.to_rdkit()
+    A_conf = molA.GetConformer()
+    B_conf = molB.GetConformer()
+    extras = dict()
+
+    for i, j in mapping.molA_to_molB.items():
+        # grab neighbours of of this pair in the mapping
+        A_nebs = [b.GetOtherAtomIdx(i)
+                  for b in molA.GetAtomWithIdx(i).GetBonds()
+                  if not b.GetOtherAtomIdx(i) in mapping.molA_to_molB]
+        B_nebs = [b.GetOtherAtomIdx(j)
+                  for b in molB.GetAtomWithIdx(j).GetBonds()
+                  if not b.GetOtherAtomIdx(j) in mapping.molA_to_molB.values()]
+
+        # for each combination of neighbours, check if they're close
+        for i, ai in enumerate(A_nebs):
+            atom_i = molA.GetAtomWithIdx(ai)
+            for j, aj in enumerate(B_nebs):
+                atom_j = molB.GetAtomWithIdx(aj)
+
+                # at least one atom must be a hydrogen, or maybe both if no element changes allowed
+                if element_change:
+                    if not (
+                            atom_i.GetAtomicNum() == 1 or atom_j.GetAtomicNum() == 1):
+                        continue
+                else:
+                    if not (
+                            atom_i.GetAtomicNum() == 1 and atom_j.GetAtomicNum() == 1):
+                        continue
+
+                dist = A_conf.GetAtomPosition(ai).Distance(
+                    B_conf.GetAtomPosition(aj))
+                if dist < atom_max_distance:
+                    extras[ai] = aj
+
+    return extras
+
+
+def get_mapping(
+        mol1, mol2, *,
+        mcs_seed=None,
+        atom_match='Any',
+        atom_match_valences=False,
+        atom_match_chiral=False,
+        atom_match_charge=False,
+        atom_ring_matches_ring=False,
+        atom_complete_rings=False,
+        atom_match_isotope=False,
+        atom_max_distance=1.0,
+):
+    """
+    Parameters
+    ----------
+    mol1, mol2 : rdkit Mol
+    mcs_seed : str, optional
+      optionally provide a starting point to improve performance over
+    atom_match : str, optional
+      one of 'Any', 'Element', 'Heavy'
+      how must elements match when constructing the mapping.  Element forces
+      an exact match, Heavy allows any heavy atoms to match, and Any allows
+      hydrogens to match heavy atoms.
+      default 'Any'
+    atom_match_distance : float, optional
+      geometric criteria for two
+    """
+    # make a copy of the explicitH version
+    mol1 = Chem.Mol(mol1)
+    mol2 = Chem.Mol(mol2)
+    # label the atoms to make mapping the implicit to explicit H version trivial
+    assign_idx(mol1)
+    assign_idx(mol2)
+    # create a copy without hydrogens
+    mol1b = Chem.RemoveHs(mol1)
+    mol2b = Chem.RemoveHs(mol2)
+
+    p = rdFMCS.MCSParameters()
+    if mcs_seed:
+        p.InitialSeed = mcs_seed
```

Comment:
if you can identify a common core ahead of time you can speed up calcs a lot here

Reply:
turned out to be more complicated, as the definition of the MCS is not as liberal as I hoped for. Maybe more setting tweaking is required.